gradio/events.py

```diff
@@ -183,13 +203,15 @@ def edit(
         built-in editor. This method can be used when this component is in a Gradio Blocks.
 
         Parameters:
-            fn: Callable function
-            inputs: List of inputs
-            outputs: List of outputs
+            fn: Callable function. When batch is False, each parameter of the function corresponds to one input component, and each return value corresponds to one output component. But if batch is True, the function must accept a *single* parameter, which is a *nested* list of inputs, where the outer list represents the batch, and the inner list represents the inputs to underlying prediction. For example, if the prediction takes two inputs, the input to the function would be [[input1, input2], [input1, input2], [input1, input2]]. Similarly, the function must return a single *nested* list of outputs. For example, if the prediction has one output, the output of the function should be [[output1], [output1], [output1]].
```

Comment:
Having the function expect a list of lists and return a list of lists is a natural api but it may get a bit clunky. For example, for a gallery it'd be a list[list[list[image]]] . Not sure if there's a better alternative.

Reply:
This is a tough one, I can't think of a better solution. I'm not sure I follow the three nested lists in your example, wouldn't it be `[[b1i1, b1i2], [b2i1, b2i2]]` for a gallery of 2 images and a batch size of 2?